gatsby-node.js

```diff
@@ -71,6 +71,37 @@ exports.createPages = ({ graphql, actions }) => {
         })
       })
     )
+    resolve(
```

Comment:
The mix of `resolve` with promises and `.then()` can be a little confusing, especially when learning. In general we use them for asynchronous operations in javascript, and you do `resolve(finalValue)` or `reject(errorThatHappened)` when the function is complete.

Once you have a `Promise` object, you can use its `then(data => { doStuffWithData ... } method to do things with the data.
```js
const promiseMaker = (resolvedVal) => {
  return new Promise((resolve, reject) => {
    resolve(resolvedVal)
  })
}

promiseMaker("one")
  .then( first => {
    promiseMaker("two")
      .then( second => {
        console.log(`got all data: ${first}, ${second}`)
      })
    })
```
If you need to make multiple calls, this is what your code should probably look like, and then call `resolve()` at the end of your code.
Because this often leads to nested code (sometimes called 'callback hell' because you get so many `then` callback functions mixed together) there is now a syntax available in node 8+: `async/await`. In order to use it you have to put `async` at the start of your function. The code below is identical to the code above:
```js

const doTheSameThingWithAsync = async () => {
  const first = await promiseMaker("one")
  const second = await promiseMaker("two")
  console.log(`got all data: ${first}, ${second}`)
}
```
You can see this recommended in the gatsby docs [here](https://www.gatsbyjs.org/docs/debugging-async-lifecycles/#use-async--await) as well.

The point is that you can use either, but:
- you correctly return one promise at the start of your code.
- wait to call resolve until the end
- use `then()` chains or `async/await` to get there
- If the last statement of any function inside is a promise, make sure you return it.

One final change I would make: 
Because `graphql()` on line 14 is already returning a promise, you could actually remove your own `new Promise` and just `return graphql(...).then(...` instead. In that case you won't need `resolve` at all!

Reply:
(The code examples above should all run correctly, even in the chrome browser console)

Reply:
I just remembered the video here, which shows the async/await bit really well: https://dev.to/ben/visual-studio-code-can-now-convert-your-long-chains-of-promisethens-into-asyncawait-automagically-1b1b